pkg/transaction/builder.go

```diff
@@ -190,15 +194,66 @@ func MakeSTXTokenTransfer(
 	tx.Auth.OriginAuth.Fee = fee.Uint64()
 	tx.Auth.OriginAuth.Nonce = nonce.Uint64()
 
-	// err = crypto.SignTransaction(tx, senderKey)
-	// if err != nil {
-	// 	return nil, fmt.Errorf("failed to sign transaction: %w", err)
-	// }
+	err = SignTransaction(tx, senderKey)
+	if err != nil {
+		return nil, fmt.Errorf("failed to sign transaction: %w", err)
+	}
 
 	return tx, nil
 }
 
-func estimateTransactionByteLength(tx TokenTransferTransaction) (int, error) {
+func MakeContractCall(
```

Comment:
This function shares a lot of logic when it comes to things like fee and nounce, They should be able to share more logic so that is only managed in one place

Reply:
Updated in https://github.com/icon-project/stacks-go-sdk/pull/17/commits/0ab00ee1b6adfd0edf542db67e13fbcbffe87304

Reply:
Nice!